### ISSUE TITLE
Line *: Char *: undefined: Deserializer

### ISSUE THREAD

**Issue by krishnanmano** [LeetCode-Feedback/LeetCode-Feedback#7229]
#### Your LeetCode username
manojkrishnan24

#### Category of the bug
- [x] Question
- [ ] Solution
- [ ] Language
- [ ] Missing Test Cases 

#### Description of the bug
Getting deserializer error for LinkedList go code which is working fine on my local machine

#### Code you used for Submit/Run operation
```
type ListNode struct {
	Val  int
	Next *ListNode
}

type MyLinkedList struct {
	size int
	head *ListNode
}


func Constructor() MyLinkedList {
	return MyLinkedList{0, nil}
}


func (this *MyLinkedList) Get(index int) int {
	if index < 0 || index >= this.size {
		return -1
	}

	node := this.head
	for i := 0; i < index+1; i++ {
		node = node.Next
	}
	return node.Val
}


func (this *MyLinkedList) AddAtHead(val int) {
	this.AddAtIndex(0, val)
}


func (this *MyLinkedList) AddAtTail(val int) {
	this.AddAtIndex(this.size, val)
}


func (this *MyLinkedList) AddAtIndex(index int, val int) {
	if index > this.size {
		return
	}

	if index < 0 {
		index = 0
	}

	if this.head == nil {
		this.head = &ListNode{Val: val, Next: nil}
	} else {
		node := this.head
		for i := 0; i < index; i++ {
			node = node.Next
		}
		node.Next = &ListNode{Val: val, Next: node.Next}
		this.size++
	}
}


func (this *MyLinkedList) DeleteAtIndex(index int) {
	if index < 0 || index >= this.size {
		return
	}

	node := this.head
	for i := 0; i < index; i++ {
		node = node.Next
	}

	node.Next = node.Next.Next
	this.size--
}
```

#### Language used for code
Go

#### Expected behavior
Run code should execute without any issue

#### Screenshots
<img width="1789" alt="Screenshot 2022-05-08 at 12 43 17 PM" src="https://user-images.githubusercontent.com/8068772/167285893-11c440aa-e330-4926-b4e3-0853fd0fda4f.png">

**Comment by LC-Yani** [comment]
Thank you for reaching out to us. I've relayed this issue to our team to investigate.

**Comment by LC-Pam** [comment]
Thank you for your time. We've submitted your feedback to the team. We appreciate your support!

**Comment by leonardchinonso** [comment]
Hi, I am having the exact same issue.

```// Node struct to represent doubly linked list node
type ListNode struct {
    Val int
    Next *ListNode
    Prev *ListNode
}

func NewListNode(val int) *ListNode {
    return &ListNode{
        Val: val,
    }
}

func (ln *ListNode) GetVal() int {
    return ln.Val
}

func (ln *ListNode) GetPrev() *ListNode {
    return ln.Prev
}

func (ln *ListNode) GetNext() *ListNode {
    return ln.Next
}

func (ln *ListNode) SetNext(node *ListNode) {
    ln.Next = node
    node.Prev = ln
}


// LinkedList struct to represent doubly linked list
type MyLinkedList struct {
    head *ListNode
    tail *ListNode
    size int
}


func Constructor() MyLinkedList {
    head := NewListNode(-1)
    tail := NewListNode(-1)
    
    head.SetNext(tail)
    
    return MyLinkedList{
        head: head,
        tail: tail,
        size: 0,
    }
}


func (this *MyLinkedList) Get(index int) int {
    if index >= this.size || index < 0 {
        return -1
    }
    
    n := index
    curr := this.head
    for n > 0 {
        curr = curr.GetNext()
        n--
    }
    
    return curr.GetVal()
}


func (this *MyLinkedList) AddAtHead(val int)  {
    node := NewListNode(val)
    
    next := this.head.GetNext()
    node.SetNext(next)
    this.head.SetNext(node)
    
    this.incSize()
}


func (this *MyLinkedList) AddAtTail(val int)  {
    node := NewListNode(val)
    
    prev := this.tail.GetPrev()
    prev.SetNext(node)
    node.SetNext(this.tail)
    
    this.incSize()
}


func (this *MyLinkedList) AddAtIndex(index int, val int)  {
    if index > this.size || index < 0 {
        return
    }
    
    if index == this.size {
        this.AddAtTail(val)
        return
    }
    
    n := index
    node := NewListNode(val)
    prev := this.head
    curr := this.head.GetNext()
    
    for n > 0 {
        prev = curr
        curr = curr.GetNext()
    }
    
    prev.SetNext(node)
    node.SetNext(curr)
    
    this.incSize()
}


func (this *MyLinkedList) DeleteAtIndex(index int)  {
    if index >= 0 && index < this.size {
        n := index
        prev := this.head
        curr := this.head.GetNext()
        for n > 0 {
            prev = curr
            curr = curr.GetNext()
        }
        prev.SetNext(curr.GetNext())
        this.decSize()
    }
}


func (this *MyLinkedList) incSize() {
    this.size++
}

func (this *MyLinkedList) decSize() {
    this.size--
}


/**
 * Your MyLinkedList object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Get(index);
 * obj.AddAtHead(val);
 * obj.AddAtTail(val);
 * obj.AddAtIndex(index,val);
 * obj.DeleteAtIndex(index);
 */```

**Comment by jadeGopher** [comment]
Hi there!
**Task Name:** 653. Two Sum IV - Input is a BST
**Link to go.playground to proof that everything works**: https://go.dev/play/p/CrQtCheQJv2
## Error message
> Line 60: Char 19: undefined: Deserializer (solution.go)
> Line 61: Char 33: undefined: Deserializer (solution.go)
> Line 104: Char 34: undefined: Serializer (solution.go)
## Code:
```
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func (t *TreeNode) Search(val int) bool {
	if t == nil {
		return false
	}

	if val == t.Val {
		return true
	}

	if val < t.Val {
		return t.Left.Search(val)
	}

	if val > t.Val {
		return t.Right.Search(val)
	}

	return false
}

func (t *TreeNode) IsSumExists(k int) bool {
	if t == nil {
		return false
	}

	term := k - t.Val

	if term < t.Val {
		if t.Left.Search(term) {
			return true
		}
	}

	if term > t.Val {
		if t.Right.Search(term) {
			return true
		}
	}

	if k <= t.Val {
		return t.Left.IsSumExists(k)
	}

	if k > t.Val {
		return t.Right.IsSumExists(k)
	}

	return false
}

func findTarget(root *TreeNode, k int) bool {
	return root.IsSumExists(k)
}
```

Can't find solution to fix that issue

**Comment by jadeGopher** [comment]
@LC-Pam are there any updates? Today I've tried submit my solution to another Task _2. Add Two Numbers_ and I got exactly the same exception: 
```
Line 50: Char 19: undefined: Deserializer (solution.go)
Line 51: Char 33: undefined: Deserializer (solution.go)
Line 94: Char 34: undefined: Serializer (solution.go)
```
Here is a [link](https://go.dev/play/p/nXZqw2ME-oj?v=goprev) to goplaygound with code that not compiles on LeetCode

**Comment by lachieggg** [comment]
Same issue.

**Comment by cyadav18** [comment]
Hi I'm facing the exactly same issue 

`type ListNode struct {
	Key  int
	Val  int
	Next *ListNode
	Prev *ListNode
}

type LinkedList struct {
	head  *ListNode
	last  *ListNode
	Size  int
	Limit int
}

func GetNewLinkedList(limit int) LinkedList {
	return LinkedList{
		head:  nil,
		last:  nil,
		Size:  0,
		Limit: limit,
	}
}

func (l *LinkedList) InsertAtStart(node *ListNode, cache map[int]*ListNode) {
	if l.head == nil {
		l.head = node
		l.last = node
	} else {
		node.Next = l.head
		l.head.Prev = node
		l.head = node
	}
	l.Size++
	if l.Size > l.Limit {
		l.RemoveLast(cache)
	}
}

func (l *LinkedList) RemoveLast(cache map[int]*ListNode) {
	if l.last == nil {
		return
	} else {
		prev := l.last.Prev
		temp := l.last
		l.last = prev
		if prev != nil {
			prev.Next = nil
		}
		delete(cache, temp.Key)
		l.Size--

	}
}

func (l *LinkedList) MoveToStart(node *ListNode) {
	if node == nil || node == l.head {
		return
	} else {
		prev := node.Prev
		next := node.Next

		node.Next = l.head
		l.head.Prev = node
		l.head = node

		prev.Next = next
		if next != nil {
			next.Prev = prev
		} else {
			l.last = prev
		}
	}
}

type LRUCache struct {
    linkedList LinkedList
	cache      map[int]*ListNode    
}


func Constructor(capacity int) LRUCache {
    return LRUCache{
		linkedList: GetNewLinkedList(capacity),
		cache:      make(map[int]*ListNode),
	}
}


func (this *LRUCache) Get(key int) int {
    if node, ok := this.cache[key]; ok {
		this.linkedList.MoveToStart(node)
		return node.Val
	}
	return -1
}


func (this *LRUCache) Put(key int, value int)  {
    if node, ok := this.cache[key]; ok {
		this.linkedList.MoveToStart(node)
		node.Val = value
	} else {
		node = &ListNode{
			Key: key,
			Val: value,
		}
		this.cache[key] = node
		this.linkedList.InsertAtStart(node, this.cache)
	}
}
`

The code works very well in my local env, 

![image](https://user-images.githubusercontent.com/25732117/209244036-d80544c4-f2ab-4dc0-b6b9-5315b812eb59.png)

can you please help here ?

**Comment by salasberryfin** [comment]
I know this is an old closed issue but, since I've seen many people asking for help over time and I faced this same problem, I just wanted to post what I've found. 

It looks like this may happen in those problems that define Go structs -maybe any custom type-. **The original code snippet has the struct definition commented out**. I usually would go to my editor, solve the problem and then come back to LeetCode and paste my solution, with the struct definition no longer commented. It looks like this is what's causing it to fail. Comment it out or remove it and it should work just fine.

Hope it helps to anyone that ends up here!
